observer/baseNodeProvider.go

```diff
@@ -32,6 +36,82 @@ func (bop *baseNodeProvider) initNodesMaps(nodes []*data.NodeData) error {
 	return nil
 }
 
+// ReloadNodes will reload the observers or the full history observers
+func (bop *baseNodeProvider) ReloadNodes(nodesType data.NodeType) data.NodesReloadResponse {
```

Comment:
add unit teste for the new functionality

Reply:
done